pkg/admin/graphql/resource.go

```diff
@@ -41,4 +42,34 @@ var nodeResource = node(
 	},
 )
 
+func init() {
+	nodeResource.AddFieldConfig("scopes", &graphql.Field{
+		Type:        connScope.ConnectionType,
+		Description: "The list of scopes for this resource.",
+		Args: relay.NewConnectionArgs(graphql.FieldConfigArgument{
+			"clientID":      &graphql.ArgumentConfig{Type: graphql.String},
+			"searchKeyword": &graphql.ArgumentConfig{Type: graphql.String},
+		}),
+		Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+			source := p.Source.(*model.Resource)
+			ctx := p.Context
+			gqlCtx := GQLContext(ctx)
+
+			// TODO(tung): Support client ID & searchKeyword filter
+			scopes, err := gqlCtx.ResourceScopeFacade.ListScopes(ctx, source.ID)
```

Comment:
When I see this, I realized we did not define an order when we return a list of scopes. We should return a list of scopes ordered by the created at ASC. Please add the ORDER BY clause to the SQL query.

Add an index `(app_id, resource_id, created_at ASC)` on `_auth_resource_scope`

Reply:
I did not change the order in this pr, but I still added the index so we can change it any time.